src/components/TransitionSwitch/TransitionSwitch.tsx

```diff
@@ -22,23 +45,35 @@ export class TransitionSwitch extends React.Component<TransitionSwitchProps, und
         const transitionProps: any = {
             ...props,
             ...{
-                key: location.pathname.split("/")[1],
+                key: this.routeProps.key,
             }
         } as any;
 
         const currentRouteProps = {
             ...this.routeProps,
-            ...{
-                location,
-            }
+            ...{location},
         };
 
+        this.setDirection(currentRouteProps.key);
```

Comment:
Updating state in render is not allowed. Use componentWillReceiveProps to relate props and state instead